app/controllers/goals_controller.rb

```diff
@@ -12,6 +12,22 @@ def create
     end
   end
 
+  def destroy
```

Comment:
:+1: Good protection of a user's goal from being deleted by a malicious user.

Now's a good time to introduce a common solution to this problem of protecting user-owned data from other logged-in users: `current_user`-scoped queries.

`current_user`-scoped queries limit the db-queried data to only the data owned by the logged-in user. Here's an example:

```
def destroy
  # Only finds the goal identified by params[:id] if it also has
  # a user_id == current_user.id:
  @goal = current_user.goals.find(params[:id])
  ...
end
```
Try using the code above and see in the `rails s` output how that changes the generated SQL query.